2017/proposals/bansal-utkarsh.md

```diff
@@ -0,0 +1,628 @@
+    alternatives are available.
+    * To run tests in parallel, there is a open source (MIT license) plugin [pytest-xdist](https://github.com/pytest-dev/pytest-xdist) that is recommended by pytest.
+* Configure TravisCI and code coverage. This will not take much time as most of the original config will run just fine. Only a
+couple of commands will have to be changed.
+
+**Challenges** The main challenge in this part is to not loose any tests in the conversion process. Code coverage data can be used to identify parts where this
+ problem occurs.
+
+
+Part 2 
+------
+This part will focus on the coordinates module and its tests. The coordinates module is a very important part of 
+MDAnalysis and any bugs in it will result in error in further calculations. Currently the coordinates module has not 
+been tested in a consistent fashion. Also all the reader/writers don’t necessarily follow the standard API.
+
+* Modify Reader/Writer API for files in the coordinates module for to follow the standard API.
+* Modify coordinates tests to use base Reader/Writer test classes. This will help to set a baseline that all 
+readers/writers will have to conform to and at the same time avoid code duplication among test classes.
+* Modify code to make use of pytest only features such as assert statements and fixtures.
+
+This part will be split into multiple pull requests, ideally one pull request for each coordinate test file.
+The following files will be fixed in this part -
+* `test_amber_inpcrd.py`
+* `test_chainreader.py`
+* `test_crd.py`
+* `test_dcd.py`
+* `test_dlpoly.py`
+* `test_dms.py`
+* `test_gms.py`
+* `test_gro.py`
+* `test_lammps.py`
+* `test_memory.py`
+* `test_mmtf.py`
+* `test_mol2.py`
+* `test_netcdf.py`
+* `test_null.py`
+* `test_pdb.py`
+* `test_pdbqt.py`
+* `test_pqr.py`
+* `test_reader_api.py`
+* `test_timestep_api.py`
+* `test_trj.py`
+* `test_trz.py`
+* `test_writer_registration.py`
+* `test_xdr.py`
+* `test_xyz.py`
+* `base.py`
+* `reference.py`
+
+This part will take a large part of my time as I have to deal with two issues at the same time - [#516](https://github.com/MDAnalysis/mdanalysis/issues/516) - Porting test
+cases to use base test classes and [#884](https://github.com/MDAnalysis/mdanalysis/issues/884) - Making changes to run with pytest. Moreover this will be the first place I
+start using pytest idioms, so that will involve a lot of discussions and reviews from my mentors.
+
+**Challenges** The main challenge in this part is to modify (if needed) the reader/writer API of the coordinates module to
+follow the API standard. Here my mentor can point out any problems with the changes via reviews. Again, this is because
+I want to do PRs on a per-file basis - it will be easier to review and get merged.
+
+
+Part 3
+------
+This part will focus on the remaining tests and will modify them to use pytest features such as assert statements and
```

Comment:
This should be part two. Switch to use as many pytest idoms as possible and work on the test performance using module wide fixtures. That has higher priority for us.